source/Symbol/SwiftASTContext.cpp

```diff
@@ -864,14 +864,31 @@ SwiftASTContext::SwiftASTContext(const SwiftASTContext &rhs)
 
   m_platform_sdk_path = rhs.m_platform_sdk_path;
 
+  // Initialize search paths and clang importer options, we need
+  // them to grab a SwiftASTContext without asserting.
+  std::vector<std::string> module_search_paths;
+  std::vector<std::pair<std::string, bool>> framework_search_paths;
+  const auto &opts = rhs.m_compiler_invocation_ap->getSearchPathOptions();
+  module_search_paths.insert(module_search_paths.end(),
+                             opts.ImportSearchPaths.begin(),
+                             opts.ImportSearchPaths.end());
+  for (const auto &fwsp : opts.FrameworkSearchPaths)
+      framework_search_paths.push_back({fwsp.Path, fwsp.IsSystem});
+  InitializeSearchPathOptions(module_search_paths, framework_search_paths);
+  GetClangImporterOptions();
+
   swift::ASTContext *lhs_ast = GetASTContext();
   swift::ASTContext *rhs_ast =
       const_cast<SwiftASTContext &>(rhs).GetASTContext();
 
-  if (lhs_ast && rhs_ast) {
-    lhs_ast->SearchPathOpts = rhs_ast->SearchPathOpts;
-  }
-  GetClangImporter();
+  // As this is a copy constructor, make sure we copy the search path
+  // opts and clang importer options from RHS to LHS.
+  assert(lhs_ast && "Invalid LHS ASTContext");
+  assert(rhs_ast && "Invalid RHS ASTContext");
+  lhs_ast->SearchPathOpts = rhs_ast->SearchPathOpts;
+
+  GetCompilerInvocation().getClangImporterOptions() =
+      rhs.m_compiler_invocation_ap->getClangImporterOptions();
```

Comment:
Isn't this redundant with the call to InitializeSearchPathOptions() and  GetClangImporterOptions() above?

Reply:
Only the copy of search paths, the second one is not because before we just initialise the importer options, we don't copy from RHS.